### PR TITLE
Take readPosition into account when copying to readLimit

### LINKED ISSUE #300: Bytes.copy, copyTo should take readPosition into account
Right now, both will copy `readRemaining()` bytes, but will start from position 0 instead of `readPosition()`, which does not make sense:

```
    @Test
    public void copyShouldCopy() {
 
        final Bytes<byte[]> src = Bytes.from("hello again");
        src.readSkip(7);
        assertThat(src.copy()).isEqualTo(src);
    }
 
    @Test
    public void copyToShouldStartFromTheReadPositionIfItUsesReadRemaining() {
 
        final Bytes<byte[]> src = Bytes.from("hello again");
        src.readSkip(7);
 
        final byte[] buffer = new byte[100];
        final int copiedLen = src.copyTo(buffer);
        assertThat(new String(buffer, 0, copiedLen)).isEqualTo(src.toString());
    }
```

### PR DESCRIPTION
**Pull request by alamar**
fixes #300

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/bytes/RandomDataInput.java [hunk 1]
```diff
@@ -313,7 +313,7 @@ default int copyTo(@NotNull byte[] bytes)
         throwExceptionIfReleased(this);
         int len = (int) Math.min(bytes.length, readRemaining());
         for (int i = 0; i < len; i++)
-            bytes[i] = readByte(start() + i);
+            bytes[i] = readByte(readPosition() + i);
```
Comment (minborg) [inline src/main/java/net/openhft/chronicle/bytes/RandomDataInput.java]: We could move out readPosition() from the inner loop.

**Comment by minborg** [comment]
These changes look right but may affect the existing code.

**Review by minborg — APPROVED** [review]

File: src/test/java/net/openhft/chronicle/bytes/BytesInternalTest.java [hunk 2]
```diff
@@ -327,6 +326,24 @@ public void testParseDouble() {
         }
     }
 
+    @Test
+    public void testCopyAfterSkip() {
+        final Bytes<byte[]> src = Bytes.from("hello again");
+
+        src.readSkip(7);
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/bytes/BytesInternalTest.java:333]: I don't understand. Doesn't readSkip increase the read position? Shouldn't `src.copy()` be "gain"?

Comment (alamar) [reply src/test/java/net/openhft/chronicle/bytes/BytesInternalTest.java:333]: I think so! But it used to be "hell" before fix.

**Pull request merged**